src/aiidalab_qe/plugins/xps/result.py

```diff
@@ -0,0 +1,247 @@
+
+    import numpy as np
+    from scipy.special import voigt_profile  # pylint: disable=no-name-in-module
+
+    result_spectra = {}
+    fwhm_voight = gamma / 2 + np.sqrt(gamma**2 / 4 + sigma**2)
+    for element, point in points.items():
+        result_spectra[element] = {}
+        final_spectra_y_arrays = []
+        total_multiplicity = sum(
+            [equivalent_sites_data[site]["multiplicity"] for site in point]
+        )
+        max_core_level_shift = max(point.values())
+        min_core_level_shift = min(point.values())
+        # Energy range for the Broadening function
+        x_energy_range = np.linspace(
+            min_core_level_shift - fwhm_voight - 1.5,
+            max_core_level_shift + fwhm_voight + 1.5,
+            500,
+        )
+        for site in point:
+            # Weight for the spectra of every atom
+            intensity = equivalent_sites_data[site]["multiplicity"]
+            relative_core_level_position = point[site]
+            y = (
+                intensity
+                * voigt_profile(
+                    x_energy_range - relative_core_level_position, sigma, gamma
+                )
+                / total_multiplicity
+            )
+            result_spectra[element][site] = [x_energy_range, y]
+            final_spectra_y_arrays.append(y)
+        total = sum(final_spectra_y_arrays)
+        result_spectra[element]["total"] = [x_energy_range, total]
+    return result_spectra
+
+
+class Result(ResultPanel):
+    title = "XPS"
+    workchain_labels = ["xps"]
+
+    def __init__(self, node=None, **kwargs):
+        super().__init__(node=node, **kwargs)
+
+    def _update_view(self):
+        import plotly.graph_objects as go
+
+        spectrum_select_prompt = ipw.HTML(
+            """
+            <div style="line-height: 140%; padding-top: 10px; padding-right: 10px; padding-bottom: 0px;"><b>
+            Select spectrum to plot</b></div>"""
+        )
+
+        voigt_profile_help = ipw.HTML(
+            """<div style="line-height: 140%; padding-top: 10px; padding-bottom: 10px">
+                Set the <a href="https://en.wikipedia.org/wiki/Voigt_profile" target="_blank">Voigt profile</a> to broaden the XPS spectra:
+            </div>"""
+        )
+
+        spectra_type = ipw.ToggleButtons(
+            options=[
+                ("Chemical shift", "chemical_shift"),
+                ("Binding energy", "binding_energy"),
+            ],
+            value="chemical_shift",
+        )
+        gamma = ipw.FloatSlider(
+            value=0.3,
+            min=0.1,
+            max=1,
+            description="Lorentzian profile (γ)",
+            disabled=False,
+            style={"description_width": "initial"},
+        )
+        sigma = ipw.FloatSlider(
+            value=0.3,
+            min=0.1,
+            max=1,
+            description="Gaussian profile (σ)",
+            disabled=False,
+            style={"description_width": "initial"},
+        )
```

Comment:
I didn't know that Python strings could use non-ASCII characters :laughing:. While these don't raise an error at the moment, I'd suggest encoding these properly just to avoid any potential problems elsewhere:
```suggestion
        gamma = ipw.FloatSlider(
            value=0.3,
            min=0.1,
            max=1,
            description="Lorentzian profile ($\gamma$)",
            disabled=False,
            style={"description_width": "initial"},
        )
        sigma = ipw.FloatSlider(
            value=0.3,
            min=0.1,
            max=1,
            description="Gaussian profile ($\sigma$)",
            disabled=False,
            style={"description_width": "initial"},
        )
```